### PR TITLE
dev(hansbug): add kaitireport support in Makefile, msmake.bat and README.md && update degrees

### PR DESCRIPTION
**Pull request by HansBug**
* Based on #258 , commit id: `6dd9ac2dbccada5aa5c45e3286aa5233b94e9ee7`
* Add support in `Makefile`, `msmake.bat`
* Add extra guide into `README.md`
* Add degrees in `sample-kaitireport.tex`

BTW, why is font color of the table-of-content part red? It is a existing rule?

### CODE REVIEW

**Comment by HansBug** [comment]
You may take a review, and try to merge the `buaathesis.cls` @sighingnow .

**Comment by huxuan** [comment]
Hi @HansBug , really appreciate the work, but it would be better to solve the conflict before sending a pull request. Let me know if you need any help.

**Comment by HansBug** [comment]
Emmmm I know that, but I failed to merge then, all of my attempts have failed. I'm not the developer of these code and not so familiar with them, so I need help to merge them. @huxuan

**Comment by HansBug** [comment]
There R 4 conflicts. The first one is easy, I guess, just take the former position.



The second one is a little bit complex, take the former position as well, but attention that the former name is `headdegree` but the newer name is `degreehead`, maybe it is the result of a refactor.



The 3rd and 4th one is structural conflict, the content is nested in a `if` statement, I will try to get rid of it later.

If I made some mistake, please notify me, thanks a lot!

**Comment by sighingnow** [comment]
I could help on merging. Thanks for the effort!

**Comment by sighingnow** [comment]
@HansBug 可以传一份开题报告的模板给我么，word的就可以~ 感谢~

**Comment by HansBug** [comment]
啊这，我这刚合并完就又有新的commit了，GG

**Comment by HansBug** [comment]
@sighingnow 我这边能要到一个去年的，邮箱给你。

**Comment by HansBug** [comment]
@soldivelot 老哥如果你这边还在开发的话，求暂停一会，这边正在进行第二轮merge 🙏 。

**Comment by HansBug** [comment]
各位，第二波合并已经完毕 @huxuan @sighingnow，准备review下吧，尽快merge进去。

**Comment by sighingnow** [comment]
> @sighingnow 我这边能要到一个去年的，邮箱给你。
> 
> 已发送到你的gmail邮箱，就是你github主页的那个，请查收。

多谢~

File: buaathesis.cls [hunk 1]
```diff
@@ -243,7 +252,7 @@ The LaTeX template for thesis of BUAA]
         {\hspace{.5em}\titlerule*[4pt]{$\cdot$}{\zihao{-4}\contentspage}}[\vspace{0.07\baselineskip}]
     \titlecontents{section}[2em]{\songti\zihao{-4}}{\thecontentslabel\ }{}
         {\hspace{.5em}\titlerule*[4pt]{$\cdot$}{\zihao{-4}\contentspage}}[\vspace{0.1\baselineskip}]
-    \titlecontents{subsection}[4em]{\vspace{-0.2\baselineskip}\songti\zihao{5}}{\thecontentslabel\ }{}
+    \titlecontents{subsection}[4em]{\vspace{-0.2\baselineskip}\heiti\zihao{5}}{\thecontentslabel\ }{}
```
Comment (sighingnow) [inline buaathesis.cls:255]: Revert this change, if it is not intended.

File: buaathesis.cls [hunk 2]
```diff
@@ -275,7 +284,7 @@ The LaTeX template for thesis of BUAA]
     bookmarksopen,
     pdftitle={BUAA thesis},
     pdfauthor={BHOSC},
-    pdfsubject={北航毕业设计论文},
+    pdfsubject={北航毕业设计开题报告/文献综述},
     pdfcreator={LaTeXed~By~BHOSC}
 }
```
Comment (sighingnow) [inline buaathesis.cls:289]: Using a if-else block here

```
\ifbuaa@ktreport
    \hypersetup{
        bookmarksnumbered,
        bookmarksopen,
        pdftitle={BUAA thesis},
        pdfauthor={BHOSC},
        pdfsubject={北航毕业设计开题报告/文献综述},
        pdfcreator={LaTeXed~By~BHOSC}
    }
\else
    \hypersetup{
        bookmarksnumbered,
        bookmarksopen,
        pdftitle={BUAA thesis},
        pdfauthor={BHOSC},
        pdfsubject={北航毕业设计论文},
        pdfcreator={LaTeXed~By~BHOSC}
    }
\fi
```

would be better

File: buaathesis.cls [hunk 3]
```diff
@@ -1190,7 +1258,20 @@ The LaTeX template for thesis of BUAA]
...
+    \def\buaa@ktclass{#1}
+}
+% \def\buaa@ktclassheadname{}
+% \def\buaa@ktclassdegreeeng{}
+% \def\buaa@ktclassheadname{}
+% \newcommand{\ktclassheadname}[3]{%
+%     \def\buaa@ktclassdegree{#1}
+%     \def\buaa@ktclassdegreeeng{#2}
+%     \def\buaa@ktclassheadname{#3}
+% }
```
Comment (sighingnow) [inline buaathesis.cls:1274]: Delete those unused definitions.

**Review by sighingnow — CHANGES_REQUESTED** [review]
Oveall LGTM. I have left some comments.

/cc @HansBug

File: buaathesis.cls (hunk 1 above)
Comment (HansBug) [reply buaathesis.cls]: Changed back to songti already

File: buaathesis.cls (hunk 2 above)
Comment (HansBug) [reply buaathesis.cls]: If statement added already.

File: buaathesis.cls (hunk 3 above)
Comment (HansBug) [reply buaathesis.cls]: Useless definitions start with ktclass already.

**Comment by HansBug** [comment]
> Oveall LGTM. I have left some comments.
> 
> /cc @HansBug

Completed, PTAL.

**Review by sighingnow — APPROVED** [review]
LGTM.

**Comment by sighingnow** [comment]
Merged.

Thanks @HansBug and @AlienMarkWong !
